Kudu.Contracts/Jobs/FunctionTemplate.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Kudu.Contracts.Jobs
+{
+    public class FunctionTemplate
+    {
+        [JsonProperty(PropertyName = "id")]
+        public string Id { get; set; }
+
+        [JsonProperty(PropertyName = "trigger")]
+        public string Trigger { get; set; }
+
+        [JsonProperty(PropertyName = "inputs")]
+        public IEnumerable<FunctionTemplateInput> Inputs { get; set; }
```

Comment:
It's not that none of them uses outputs, but more that the selection of templates in the UI currently is based on the trigger type and language rather than anything else. The whole template story is very shaky hence the shaky APIs. I think we would need to move to sone sort of auto generated templates that cover all the possible combinations of inputs, triggers, and outputs? Maybe?